config/fi_check_package.m4

```diff
@@ -130,7 +130,7 @@ AC_DEFUN([_FI_CHECK_PACKAGE_LIB], [
                           unset fi_Lib])])])])
 
     AS_IF([test "$fi_check_package_lib_happy" = "yes"],
-          [$1_LIBS="-l$2 $4"
+          [$1_LIBS="-l$2 $4 $$1_LIBS"
```

Comment:
Well, a provider *could* call it twice if there were multiple packages it was checking, unless it is documented that this macro is call once only.